src/main/scala/com/microsoft/hyperspace/index/rankers/JoinIndexRanker.scala

```diff
@@ -16,40 +16,80 @@
 
 package com.microsoft.hyperspace.index.rankers
 
-import com.microsoft.hyperspace.index.IndexLogEntry
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import com.microsoft.hyperspace.index.{IndexLogEntry, IndexLogEntryTags}
+import com.microsoft.hyperspace.util.HyperspaceConf
 
 /**
- * Ranker class for Join rule indexes
+ * Ranker class for Join rule indexes.
  */
 object JoinIndexRanker {
 
   /**
    * Rearranges all available index options according to their cost. The first one is the best
    * with minimum cost.
    *
-   * Pick the one with least amount of shuffling and most number of buckets. When two indices have
-   * same number of buckets, there is zero shuffling. If number of buckets differ, one of the
-   * indices gets reshuffled into the number of buckets equal to the other. Secondly, more the
-   * number of buckets, better the parallelism achieved during join in general, assuming there is
-   * no resource constraint.
+   * If hybridScanEnabled is false, pick the one with least amount of shuffling and most number
+   * of buckets. When two indices have same number of buckets, there is zero shuffling.
+   * If number of buckets differ, one of the indices gets reshuffled into the number of buckets
+   * equal to the other.
+   * Secondly, more the number of buckets, better the parallelism achieved during join in general
+   * assuming there is no resource constraint.
+   *
+   * If hybridScanEnabled is true, rank algorithm follows the algorithm above, but we prioritize
+   * the index with larger and usable index data for each index, so as to minimize the amount of
```

Comment:
Is this sentence written correctly?
"we prioritize the index with larger and usable index data for __each index__"
Should not the bold part be "for each Join child"?